lesson3.md

```diff
@@ -0,0 +1,841 @@
+
+```javascript
+true  || true   // true
+true  || false  // true
+false || true   // true
+false || false  // false
+```
+
+
+---
+
+
+
+### Example - OR ||
+
+To apply to a job, I need to know German OR English
+
+```js
+let knowsGerman = false;
+let knowsEnglish = false;
+
+let canApply = knowsGerman || knowsEnglish;
+console.log(canApply); // ?
+
+```
+
+false!
+<!-- .element: class="fragment" -->
+
+
+---
+
+
+### Let's practice
+
+Use JavaScript to compute:
+
+* Am I the youngest person in the group?
+* Am I younger than at least one person in the group?
+
+
+The group is you, the person to your left and the one to your right.
+
+
+---
+
+
+### Solution
+
+Lets assume my age is 42.
+
+Am I the youngest person in the group?
+
+I need to be younger than person to my left AND I need to be younger than person to my right
+
+```js
+(42 < 20) && (42 < 35)  // false - I am not the youngest person
+```
+
+
+---
+
+
+Am I younger than at least one person in the group?
+
+I need to be younger than person to my left OR I need to be younger than person to my right
+
+```js
+(42 < 20) || (42 < 35)
+// false - I am not younger than at least one other person
+```
+
+
+---
+
+
+### Logical NOT !
+
+Logical NOT negates a boolean:
+
+```javascript
+!true  // false
+!false // true
+```
+
+Example: Am I NOT the youngest person?
+
+
+---
+
+
+### Operator precedence
+
+What do you think is the result?
+
+```js
+2 + 2 * 2
+```
+
+Result: `6` <!-- .element: class="fragment" -->
+
+
+---
+
+
+* Operators have priorities
+* Operators with higher precedence are evaluated first.
+* For example: multiplication `*` is more important than addition `+`
+* [You can see the full list here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence)
+
+
+---
+
+
+Parentheses have the highest precedence and always win:
+
+```js
+(2 + 2) * 2   // 8
+2 + (2 * 2)   // 6
+```
+
+Always use parentheses when chaining!
+
+It makes your code more clear.
+
+
+---
+
+
+### Quiz
+
+What does the following return?
+
+```js
+(false && false) || true
+```
+
+<br/>
+
+
+```js
+(false && false) || true   // false && false is false
+false || true              // returns true
+```
+<!-- .element: class="fragment" -->
+
+
+---
+
+
+### TIP: variables instead of chaining
+
+Which is more understandable?
+
+```js
+42 > 22 && 42 > 35
+```
+
+Or
+
+```js
+let myAge = 42;
+let leftPersonAge = 22;
+let rightPersonAge = 35;
+
+let olderThanLeft = myAge > leftPersonAge;
+let olderThanRight = myAge > rightPersonAge;
+let isOldest = olderThanLeft && olderThanRight;
+```
+
+
+---
+
+
+### Quick Quiz, now in VSCode!
+
+Let's say, everyone here wants to go to the cinema.
+
+How many people in total are going to the cinema?
+<!-- .element: class="fragment" -->
+
+A ticket to watch the movie *The Joker* costs 8€, how much do we have to pay in total?
+<!-- .element: class="fragment" -->
+
+Saskia and I volunteered to pay for the tickets, how much each one pays?
```

Comment:
Saskia and I volunteered to pay for the tickets. How much do we each have to pay?